### PR TITLE
Extracts a ton of code into individual files under a new dir 'source/js/foundation'

### PR DESCRIPTION
**Pull request by mmmavis**
Related to #4313 

This PR only extracts some code in `main.js` into individual files under a new dir `source/js/foundation`. More code optimization & cleanup to follow in new PRs.

### CODE REVIEW

File: source/js/foundation/inject-react/multipage-nav.js [hunk 1]
```diff
@@ -0,0 +1,36 @@
+import React from "react";
```
Comment (mmmavis) [inline source/js/foundation/inject-react/multipage-nav.js]: https://github.com/mozilla/foundation.mozilla.org/pull/4433/files#diff-bcb353d4a212e3c287b0d8d8c377f1c5L296-L321

File: source/js/foundation/inject-react/news.js [hunk 2]
```diff
@@ -0,0 +1,21 @@
+import React from "react";
```
Comment (mmmavis) [inline source/js/foundation/inject-react/news.js]: https://github.com/mozilla/foundation.mozilla.org/pull/4433/files#diff-bcb353d4a212e3c287b0d8d8c377f1c5L323-L333

File: source/js/foundation/template-js-handler/global/sticky-cta.js [hunk 3]
```diff
@@ -0,0 +1,7 @@
+/**
+ * Bind handler to
+ *
+ */
+export default () => {
+  // to be added
```
Comment (mmmavis) [inline source/js/foundation/template-js-handler/global/sticky-cta.js]: to be added in the next PR

File: source/js/foundation/inject-react/multipage-nav.js (hunk 1 above)
Comment (mmmavis) [inline source/js/foundation/inject-react/multipage-nav.js]: https://github.com/mozilla/foundation.mozilla.org/pull/4433/files#diff-bcb353d4a212e3c287b0d8d8c377f1c5L297-L321

File: source/js/foundation/inject-react/news.js (hunk 2 above)
Comment (mmmavis) [inline source/js/foundation/inject-react/news.js]: https://github.com/mozilla/foundation.mozilla.org/pull/4433/files#diff-bcb353d4a212e3c287b0d8d8c377f1c5L323-L333

File: source/js/foundation/inject-react/pulse-project-list.js [hunk 4]
```diff
@@ -0,0 +1,31 @@
+import React from "react";
```
Comment (mmmavis) [inline source/js/foundation/inject-react/pulse-project-list.js]: https://github.com/mozilla/foundation.mozilla.org/pull/4433/files#diff-bcb353d4a212e3c287b0d8d8c377f1c5L335-L355

File: source/js/foundation/inject-react/petition.js [hunk 5]
```diff
@@ -0,0 +1,37 @@
+import React from "react";
```
Comment (mmmavis) [inline source/js/foundation/inject-react/petition.js]: https://github.com/mozilla/foundation.mozilla.org/pull/4433/files#diff-bcb353d4a212e3c287b0d8d8c377f1c5L269-L294

File: source/js/foundation/inject-react/share-button-group.js [hunk 6]
```diff
@@ -0,0 +1,23 @@
+import React from "react";
```
Comment (mmmavis) [inline source/js/foundation/inject-react/share-button-group.js]: https://github.com/mozilla/foundation.mozilla.org/pull/4433/files#diff-bcb353d4a212e3c287b0d8d8c377f1c5L357-L371

File: source/js/foundation/template-js-handler/internet-health-see-more.js [hunk 7]
```diff
@@ -0,0 +1,23 @@
+/**
```
Comment (mmmavis) [inline source/js/foundation/template-js-handler/internet-health-see-more.js]: https://github.com/mozilla/foundation.mozilla.org/pull/4433/files#diff-bcb353d4a212e3c287b0d8d8c377f1c5L227-L244

File: source/js/foundation/template-js-handler/load-more-entries.js [hunk 8]
```diff
@@ -0,0 +1,49 @@
+/**
```
Comment (mmmavis) [inline source/js/foundation/template-js-handler/load-more-entries.js]: https://github.com/mozilla/foundation.mozilla.org/pull/4433/files#diff-bcb353d4a212e3c287b0d8d8c377f1c5L466-L509

File: source/js/foundation/template-js-handler/participate-page-donate.js [hunk 9]
```diff
@@ -0,0 +1,19 @@
+/**
```
Comment (mmmavis) [inline source/js/foundation/template-js-handler/participate-page-donate.js]: https://github.com/mozilla/foundation.mozilla.org/pull/4433/files#diff-bcb353d4a212e3c287b0d8d8c377f1c5L246-L258

File: source/js/foundation/template-js-handler/pulse-profile-card.js [hunk 10]
```diff
@@ -0,0 +1,80 @@
+import { ReactGA } from "../../common/react-ga-proxy.js";
```
Comment (mmmavis) [inline source/js/foundation/template-js-handler/pulse-profile-card.js]: https://github.com/mozilla/foundation.mozilla.org/pull/4433/files#diff-bcb353d4a212e3c287b0d8d8c377f1c5L390-L464

File: source/js/foundation/template-js-handler/pulse-profile-list-filter.js [hunk 11]
```diff
@@ -0,0 +1,22 @@
+import { ReactGA } from "../../common/react-ga-proxy.js";
```
Comment (mmmavis) [inline source/js/foundation/template-js-handler/pulse-profile-list-filter.js]: https://github.com/mozilla/foundation.mozilla.org/pull/4433/files#diff-bcb353d4a212e3c287b0d8d8c377f1c5L373-L388

File: source/js/foundation/inject-react/multipage-nav.js [hunk 12]
```diff
@@ -0,0 +1,35 @@
...
+
+/**
+ * Inject multipage navs (secondary navs)
+ * @param {Array} apps The existing array we are using to to track all ReactDOM.render calls
+ */
+export default apps => {
+  if (document.querySelector(`#multipage-nav`)) {
+    let links = [];
+
+    links = [].map.call(
```
Comment (Pomax) [inline source/js/foundation/inject-react/multipage-nav.js]: omg we need to rewrite this later so badly =D

Comment (mmmavis) [reply source/js/foundation/inject-react/multipage-nav.js]: yea I will make sure this gets upated! lol

**Comment by mmmavis** [comment]
Yay!

**Pull request merged**